### ISSUE TITLE
fluent-bit does not repsect the DB path in the input section

### ISSUE THREAD

**Issue by yiwenshao** [fluent/fluent-bit#2245]
## Bug Report

**Describe the bug**
I am using fluent-bit 1.14 with the following configuration:
```
[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log1
    DB /fluentbit-pos/pods
    Tag rule1.*

[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log2
    DB /fluentbit-pos/pods
    Tag rule3.*

[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log3
    DB /fluentbit-pos/pods2
    Tag rule2.*

[OUTPUT]
    Name  stdout
    Match *
```

I do find log in the output. However, under directory /fluentbit-pos, only one pos file **pods** is generated. Where I inspect the content using sqlite, I find that /fluentbit-pos/pos contains offset info of /data/log1,/data/log2, and /data/log3.

**Expected behavior**
/fluentbit-pos/pods should only contain offset info of  /data/log1 and /data/log2. There should be another file /fluentbit-pos/pod2 which tracks /data/log3.

----

Also, I try to change my configuration to 

```
[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log1
    DB /fluentbit-pos/pods
    Tag rule1.*

[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log2
    DB /fluentbit-pos/pods3
    Tag rule3.*

[INPUT]
    Name tail
    storage.type  filesystem
    Path /data/log3
    DB /fluentbit-pos/pods2
    Tag rule2.*

[OUTPUT]
    Name  stdout
    Match *
```
This time, fluentbit work as expectd. three files are generated under /fluentbit-pos. /fluentbit-pos/pods, /fluentbit-pos/pods3, /fluentbit-pos/pods2 tracks /data/log1, /data/log2, /data/log3 respectively.

**Comment by edsiper** [comment]
We don't use a ```pos``` file as Fluentd does, we use SQLite as a backend database. If you set the same database path, it will be shared. Just set different database files for different sections if that is desired.

**Comment by yiwenshao** [comment]
Thanks for your reply.
The issue is that I set different database paths, but the database file is still shared.
The first configuration above contains three input sections, the first and second section share database path /fluentbit-pos/pods and

**Comment by yiwenshao** [comment]
Thanks for your reply.
The issue is that I set different database paths, but the database file is still shared.
The first configuration above contains three input sections, the first and second section share database path /fluentbit-pos/pods and the third input section has database path  /fluentbit-pos/pods, so I expect the database files to be created. However, only one database file  /fluentbit-pos/pods is created, and this file traces /data/log1, /data/log2, and /data/log3. This is strange because I expect /fluentbit-pos/pods to trace only /data/log1, and /data/log2. There should be another database file /fluentbit-pos/pods2, which tracks /data/log3.

![image](https://user-images.githubusercontent.com/10302159/84336964-0d6a4300-abcb-11ea-9f2e-528fa1c48776.png)

In the second configuration I specified three database files /fluentbit-pos/pods /fluentbit-pos/pods2 and /fluentbit-pos/pods3. 
![image](https://user-images.githubusercontent.com/10302159/84337151-6cc85300-abcb-11ea-917d-7b8e6786d7d6.png)

This time, three database files are generated, and fluent-bit work as expected.
